app/components/markdown.tsx

```diff
@@ -100,14 +100,20 @@ export function PreCode(props: { children: any }) {
 }
 
 function escapeDollarNumber(text: string): string {
-  let isInCode = false;
+  let isInMultilineCode = false;
   return text.split('\n').map(line => {
     if (line.trim() === '```' || line.trim() === '`') {
-      isInCode = !isInCode;
+      isInMultilineCode = !isInMultilineCode;
       return line;
     }
-    if (!isInCode) {
-      return line.replace(/(?<!\\)\$\d+([,.](\d+[,.])?\d+)?(?!.*\$\B)(?!`+)/g, '\\$&');
+    if (!isInMultilineCode) {
```

Comment:
it's better

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/7174fd99-437b-4052-8a64-2fdb8f9eb900)


I think it should be fine for inside a table

Reply:
I just made another edits. The previous one had issues rendering latex. Check the new one. Hopefully there are no more issues.